oracle-server/src/lib/octokit.ts

```diff
@@ -24,3 +43,86 @@ export function getOctokit() {
 
   return _octokit;
 }
+
+export function getGraphql() {
+  return getOctokit().graphql.defaults({
+    request: {
+      hook: getAppAuth().hook,
+    },
+  });
+}
+
+interface QueryData {
+  repository: {
+    pullRequest: {
+      merged: boolean;
+      state: "MERGED" | "OPEN" | "CLOSED";
+      author: { login: string; url: string };
+      closingIssuesReferences: {
+        nodes: {
+          number: number;
+          url: string;
+          assignees: {
+            nodes: {
+              resourcePath: string;
+              name: string;
+            }[];
+          };
+          state: "OPEN" | "CLOSED";
+          closed: boolean;
+          stateReason: "COMPLETED" | "REOPENED" | "NOT_PLANNED";
+        }[];
+      };
+    };
+  };
+}
+
+export async function queryPullRequest({
+  owner,
+  repo,
+  pull_number,
+}: GitHubResourceParams<"pull">): Promise<QueryData> {
+  try {
+    const data = await getGraphql()(
+      `
+query pullRequest($name: String!, $owner: String!, $num: Int!){
+  repository(name: $name, owner: $owner) {
+    pullRequest(number: $num) {
+      merged
+      state
+      author {
+        __typename
+        login
+        url
+      }
+      closingIssuesReferences (first:1) {
+        nodes {
+          number
+          url
+          assignees (first:1) {
+            nodes {
+              resourcePath
+              name
+            }
+          }
```

Comment:
why no `url` for `assignees`?

Reply:
resourcePath is `/${username}`, so this was easier to parse

Reply:
Ideally, we use `id`. There may be a way to get it via a relation but didn't did too hard

Reply:
hm. if we use `url` for both PR author and issue assignee, there's no need to parse anything, right?